### DISCUSSION TITLE
Authentication jwt and PostGUI

### DISCUSSION THREAD

**Discussion by bmgante** [PostgREST/postgrest#2461] [Q&A] [answered]
Hi all,

I am trying to setup a PostGUI (https://github.com/priyank-purohit/PostGUI) which makes use of postgrest but getting struggled with jwt authentication. 

Does anyone here has some kind of experience with this app? I did the config as mentioned in POSTGUI read me but when i click on login button nothing happens... at postgrest logs i can see that the post /rpc/login succeeds but at postgui application level nothing happens (and there are no logs!).

Example of a manual post to /rpc/login:

```
TRUT10236:network-catalog-ui-api bruno.gante$ curl -X 'POST' 'http://127.0.0.1:3000/rpc/login' -H 'accept: application/json' -H 'Content-Type: application/json' -d '{"email": "admin@domain.com", "pass": "CHNAGME123"}'
{"token":"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJyb2xlIjoiZWRpdHVzZXIiLCJlbWFpbCI6ImFkbWluQGRvbWFpbi5jb20iLCJleHAiOjE2NjIxMjMwNjd9.INmxG3p0k5rFvOqfzh0pqsNmUGw6TYaZvJPAWTXeT3o"}
```

Example of postgrest log when i try to login in PostGUI with valid credentials:

`172.18.0.1 - - [02/Sep/2022:11:38:53 +0000] "POST /rpc/login HTTP/1.1" 200 - "http://localhost:8771/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36"`

Example of postgrest log when i try to login in PostGUI with invalid credentials:

`172.18.0.1 - - [02/Sep/2022:11:59:22 +0000] "POST /rpc/login HTTP/1.1" 403 - "http://localhost:8771/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36"`

Is there someone here that could help please?

Thanks

**Comment by bmgante** [comment]
![image](https://user-images.githubusercontent.com/27358817/188154148-1fd0e200-ff68-41a9-92a2-a760929e8272.png)
![image](https://user-images.githubusercontent.com/27358817/188154319-1640fe6a-eb32-4a35-b0d8-fdd2cf152104.png)

**Comment by bmgante** [comment]
Below my PostGUI config. It is running in docker. Did not enabled https via nginx… is it really necessary to get editPanel available and put authentication working? 

/PostGUI/src/data # cat config.json
{
"databases": [
{
"title": "Network Catalog",
"url": "http://localhost:3000/",
"publicDbAcessType": "private",
"foreignKeySearch": true,
"primaryKeyFunction": true,
"regexSupport": false
}
],
"logoUrl": null,
"seq_column_names": [],
"errorMsg": "ERROR"
}

**Comment by wolfgangwalther** [comment, marked as answer]
This seems to be an issue with PostGUI as you have identified yourself... in the cross-posted thread at https://github.com/priyank-purohit/PostGUI/issues/56#issuecomment-1238474541.
